### ISSUE TITLE
0pdd didn't close GitHub ticket

### ISSUE THREAD

**Issue by g4s8** [yegor256/0pdd#205]
`@todo` puzzle was removed from source code, by 0pdd didn't close a ticket https://github.com/zerocracy/farm/issues/703
also this puzzle (`258-bca58f1a`) was not removed from page http://www.0pdd.com/p?name=zerocracy/farm
but there are no such puzzle in fresh master branch (checked latest commit hash), can't find it with text-search and `pdd` tool can't find this puzzle also. Here is the result of `pdd --source=$(pwd) --verbose --file=/tmp/pdd.xml`:
```xml
<?xml version="1.0"?>
<?xml-stylesheet type='text/xsl' href='http://pdd-xsl.teamed.io/0.20.xsl'?>
<puzzles xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://pdd-xsd.teamed.io/0.20.xsd" version="0.20" date="2018-05-20T08:01:52Z">
  <puzzle>
    <ticket>62</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>62-6eaaf140</id>
    <lines>31-36</lines>
    <body>We have to implement simple slack session object to use inside 'notify_in_slack' stakeholder-script and fake object for tests. This session object may wrap `SlackSession` from `com.ullink.slack.simpleslackapi`. After that we should get rid of `Mockito` in this test.</body>
    <file>src/test/resources/com/zerocracy/bundles/notify_in_slack_without_user/_before.groovy</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2017-07-18T21:22:37Z</time>
  </puzzle>
  <puzzle>
    <ticket>447</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>447-38b7b52b</id>
    <lines>27-29</lines>
    <body>register_impediment stakeholder should pay attention to impediment count requested by user and not allow to register new impediments if threshold has been reached. See #447 comments for details.</body>
    <file>src/test/resources/com/zerocracy/bundles/impediments/_after.groovy</file>
    <author>Yegor Bugayenko</author>
    <email>yegor256@gmail.com</email>
    <time>2018-03-04T07:45:01Z</time>
  </puzzle>
  <puzzle>
    <ticket>889</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>889-ebed2d87</id>
    <lines>23-27</lines>
    <body>Implement stakeholder 'check_automatic_vacation.groovy'. We should implement auto-vacation mechanism. See https://github.com/zerocracy/farm/issues/889 and http://www.zerocracy.com/policy.html#52 for details. When done uncomment the assertion below to enable this test.</body>
    <file>src/test/resources/com/zerocracy/bundles/turn_on_vacation_for_too_many_negative_awards/_after.groovy</file>
    <author>Carlos Miranda</author>
    <email>carlos.m.a.miranda@accenture.com</email>
    <time>2018-04-19T04:19:00Z</time>
  </puzzle>
  <puzzle>
    <ticket>510</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>510-7cdf5e0a</id>
    <lines>23-27</lines>
    <body>Let's implement a test for small-size pull requests. At moment it's not possible because it requires to construct specific json for PR with fields 'additions' and 'deletions' via MkStorage. Also `add_new_pull_request_to_wbs` should be fixed. See https://developer.github.com/v3/pulls/#get-a-single-pull-request</body>
    <file>src/test/resources/com/zerocracy/bundles/small_pr_skip_wbs/_before.groovy</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-03-06T21:49:08Z</time>
  </puzzle>
  <puzzle>
    <ticket>511</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>511-1a49e160</id>
    <lines>129-132</lines>
    <body>S3Item does not handle InterruptedException properly. It should not allow reading from io.channel stream if thread was interrupted because it automatically check thread status before reading and can throw `ClosedChannelException`.</body>
    <file>src/test/java/com/zerocracy/farm/S3ItemTest.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-02-02T19:41:08Z</time>
  </puzzle>
  <puzzle>
    <ticket>476</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>476-cbedeee7</id>
    <lines>124-127</lines>
    <body>Method freeze should wait for all processed claims, and stop giving new claims with IOException, it should block all SyncProjects so they stops to return SyncItems and wait for all current opened SyncItems to close.</body>
    <file>src/test/java/com/zerocracy/farm/sync/SyncFarmTest.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-04-23T07:35:02Z</time>
  </puzzle>
  <puzzle>
    <ticket>552</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>552-b158b0f7</id>
    <lines>41-43</lines>
    <body>Add tests for details(), links(), wallet() and rate() in People.java because these methods are not fully covered. Then remove People.java from jacoco excludes in pom.xml</body>
    <file>src/test/java/com/zerocracy/pmo/PeopleTest.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-03-12T10:22:09Z</time>
  </puzzle>
  <puzzle>
    <ticket>166</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>166-540fefdb</id>
    <lines>26-28</lines>
    <body>let's implement this stakeholder. It will update precedences.xml. The update should occur whenever a GitHub ticket gets assigned to a milestone.</body>
    <file>src/main/resources/com/zerocracy/stk/pm/staff/milestones/update_precedences.groovy</file>
    <author>Carlos Miranda</author>
    <email>carlosmiranda@users.noreply.github.com</email>
    <time>2017-08-08T08:49:03Z</time>
  </puzzle>
  <puzzle>
    <ticket>535</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>535-b70e915c</id>
    <lines>35-39</lines>
    <body>Every time we close a job completed by a developer, we should count the amount of messages that person posted in the GitHub issue or a pull request. Let's count not only issue messages, but also commit messages. Then, post it to verbosity.xml</body>
    <file>src/main/resources/com/zerocracy/stk/pm/in/orders/close_job.groovy</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-03-07T19:08:04Z</time>
  </puzzle>
  <puzzle>
    <ticket>835</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>835-3f8bfc99</id>
    <lines>38-41</lines>
    <body>At moment user.exists() throws NullPointerException for tests, see https://github.com/jcabi/jcabi-github/issues/1370 After jcabi bug fix this condition should be removed and test assertion in `remove_stale_users/_after.groovy` should be uncommented.</body>
    <file>src/main/resources/com/zerocracy/stk/pmo/people/remove_stale_users.groovy</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-03-20T13:29:53Z</time>
  </puzzle>
  <puzzle>
    <ticket>492</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>492-2cc36189</id>
    <lines>26-30</lines>
    <body>Let's implement this stakeholder. It will take a random user from the list, which has the oldest updated attribute, go to its GitHub account, fetch all repositories he owns and contributes to and fetch the most popular languages from them. GitHub provides that information.</body>
    <file>src/main/resources/com/zerocracy/stk/pmo/people/update_skills.groovy</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-02-02T16:15:10Z</time>
  </puzzle>
  <puzzle>
    <ticket>696</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>696-65091594</id>
    <lines>38-40</lines>
    <body>Add test for pay_all_depts stakeholder script. It is not possible because payments are not testable. It should be fixed in #565 ticket.</body>
    <file>src/main/resources/com/zerocracy/stk/pmo/debts/pay_all_debts.groovy</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-04-02T13:54:17Z</time>
  </puzzle>
  <puzzle>
    <ticket>552</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>552-9d3ef349</id>
    <lines>46-49</lines>
    <body>Let's keep agenda inside `people.xml` and update it when agenda changed as described in https://github.com/zerocracy/farm/issues/366#issuecomment-359568311 Also put it as 'agenda' element in TkGang xml.</body>
    <file>src/main/java/com/zerocracy/pmo/People.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-03-10T18:17:34Z</time>
  </puzzle>
  <puzzle>
    <ticket>715</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>715-1243e0dc</id>
    <lines>24-27</lines>
    <body>Change 'com.zerocracy.pmo' classes constructor to accept `Pmo` instead of `Project`, also all tests and groovy scripts should be fixed after refactoring. It is: Agenda, Catalog, People, Projects, Rfps, Speed.</body>
    <file>src/main/java/com/zerocracy/pmo/package-info.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-04-05T14:17:58Z</time>
  </puzzle>
  <puzzle>
    <ticket>925</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>925-41a80871</id>
    <lines>33-35</lines>
    <body>Vacancies are not rendered on `/vacancies` page. This page should list all current vacancies in all projects. Let's display vacancy author, date and text.</body>
    <file>src/main/java/com/zerocracy/pmo/Vacancies.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-04-20T10:39:05Z</time>
  </puzzle>
  <puzzle>
    <ticket>925</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>925-74129afd</id>
    <lines>36-38</lines>
    <body>Old vacancies are not removed automatically. Let's add new stakeholder which will remove vacancies from `vacancies.xml` which are older than 32 days.</body>
    <file>src/main/java/com/zerocracy/pmo/Vacancies.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-04-23T06:45:03Z</time>
  </puzzle>
  <puzzle>
    <ticket>703</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>703-5f3e27dc</id>
    <lines>33-36</lines>
    <body>Options are not used. Election should check maxJobsInAgenda option, notify stakeholders should check notify options: notifyPublish, notifyRfps, notifyStudents. Also user should be able to download, edit and upload options.xml file on profile page.</body>
    <file>src/main/java/com/zerocracy/pmo/Options.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-04-09T11:56:43Z</time>
  </puzzle>
  <puzzle>
    <ticket>422</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>422-1711e986</id>
    <lines>38-43</lines>
    <body>Title element, of an order in agenda, has been declared in agenda.xsd. Let's add method title(...), which will set the title of on order and add a new stakeholder, set_agenda_title_from_github.groovy, which will get title of a job from GitHub and set it to Agenda. THe reason for the new stakeholder is that not all orders will come from Github, some may also come from Jira, Trello etc, in the future.</body>
    <file>src/main/java/com/zerocracy/pmo/Agenda.java</file>
    <author>amihaiemil</author>
    <email>amihaiemil@gmail.com</email>
    <time>2018-03-19T18:36:25Z</time>
  </puzzle>
  <puzzle>
    <ticket>400</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>400-fba0df9b</id>
    <lines>32-35</lines>
    <body>Use this class, to periodically update contents from ./heapdump.hprof file to S3. Should be implemented after #680 would be resolved as part of the routine work, that needs to be done in background.</body>
    <file>src/main/java/com/zerocracy/entry/HeapDump.java</file>
    <author>Izbassar Tolegen</author>
    <email>t.izbassar@gmail.com</email>
    <time>2018-03-06T19:17:38Z</time>
  </puzzle>
  <puzzle>
    <ticket>400</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>400-cf11083e</id>
    <lines>80-84</lines>
    <body>Should return the latest version of heap dump, available in S3 or temporary storage (if it is up to date). Cover implementation with unit and integration tests.</body>
    <file>src/main/java/com/zerocracy/entry/HeapDump.java</file>
    <author>Izbassar Tolegen</author>
    <email>t.izbassar@gmail.com</email>
    <time>2018-03-06T19:17:38Z</time>
  </puzzle>
  <puzzle>
    <ticket>400</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>400-1ca94fd4</id>
    <lines>94-98</lines>
    <body>Should save heap dump to S3 bucket if file with dump exists, but only do that, if there is new version of file available. Ignore, if we already have latest version in S3. Should be covered with unit and integration tests.</body>
    <file>src/main/java/com/zerocracy/entry/HeapDump.java</file>
    <author>Izbassar Tolegen</author>
    <email>t.izbassar@gmail.com</email>
    <time>2018-03-06T19:17:38Z</time>
  </puzzle>
  <puzzle>
    <ticket>489</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>489-f307c752</id>
    <lines>54-57</lines>
    <body>Change claim.xsl in such a way that it displays the children of this claim as well. They are returned in the children element and represent all the claims that have the cause equal to this claim's id.</body>
    <file>src/main/java/com/zerocracy/tk/project/TkClaim.java</file>
    <author>amihaiemil</author>
    <email>amihaiemil@gmail.com</email>
    <time>2018-03-02T17:57:54Z</time>
  </puzzle>
  <puzzle>
    <ticket>425</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>425-40a20d3d</id>
    <lines>40-42</lines>
    <body>Download and analyze heap dump from https://www.0crat.com/heapdump after next out of memory error, then fix detected memory leaks. Also waiting for #400 to be solved.</body>
    <file>src/main/java/com/zerocracy/tk/TkDump.java</file>
    <author>amihaiemil</author>
    <email>amihaiemil@gmail.com</email>
    <time>2018-03-03T16:29:23Z</time>
  </puzzle>
  <puzzle>
    <ticket>400</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>400-9e938816</id>
    <lines>43-45</lines>
    <body>Should use HeapDump to get contents from S3, rather than from the server file, as it might be deleted and won't be available anymore.</body>
    <file>src/main/java/com/zerocracy/tk/TkDump.java</file>
    <author>Izbassar Tolegen</author>
    <email>t.izbassar@gmail.com</email>
    <time>2018-03-06T19:17:38Z</time>
  </puzzle>
  <puzzle>
    <ticket>653</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>653-b9d34fcf</id>
    <lines>40-43</lines>
    <body>We should allow user to send `/join` requests only once a week, it can be checked and updated via People.applied methods. The 7 days value can be accessed via Par after https://github.com/zerocracy/zerocracy.github.io/issues/9 fix.</body>
    <file>src/main/java/com/zerocracy/tk/TkJoin.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-04-01T10:14:09Z</time>
  </puzzle>
  <puzzle>
    <ticket>800</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>800-10ded7b0</id>
    <lines>44-55</lines>
    <body>Requests to /join should add an entry to 'resumes.xml' then an 'examiner' should be assigned. When the examiner "invites" a user, we check whether there was a resume for that user. If yes, we pay examiner +32 points. Examiner can reject a resume by saying 'deny {username}' (where username is a login in 'resumes.xml') to Zerocrat. In that case we also pay +32 to the examiner. Each user should see the status of his/her resume at /join. If the resume is there, he/she should see the resume, not the form. If he/she already has a mentor, there should be a redirect, saying that "User @yegor256 is already your mentor, no need to join again". See https://github.com/zerocracy/farm/issues/800#issuecomment-375551970 comment for details.</body>
    <file>src/main/java/com/zerocracy/tk/TkJoin.java</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-03-26T09:35:30Z</time>
  </puzzle>
  <puzzle>
    <ticket>474</ticket>
    <estimate>30</estimate>
    <role>DEV</role>
    <id>474-ba887765</id>
    <lines>621-623</lines>
    <body>branch coverage, complexity coverage and class coverage limits are disabled, because current coverage is too low. Let's add tests and increase class and branch coverage.</body>
    <file>pom.xml</file>
    <author>g4s8</author>
    <email>g4s8.public@gmail.com</email>
    <time>2018-02-20T16:56:22Z</time>
  </puzzle>
</puzzles>
```

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C3T46CUJJ) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by 0crat** [comment]
Job `gh:yegor256/0pdd#205` already assigned to @yegor256, can't assign to @yegor256

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by yegor256** [comment]
@g4s8 I think this problem is related to #204 We fail on timeouts, I believe. I increased the timeout to 240 seconds. Let's push something to the `zerocracy/farm` repo and see what happens. I will push now.

**Comment by yegor256** [comment]
@g4s8 I suspect it's not the delay, but something else, since 240 seconds is more than enough to parse the repo

**Comment by g4s8** [comment]
@yegor256 it looks like `zerocracy/farm` repo is too big for pdd:
> "set -x && set -e && set -o pipefail && cd /tmp/0pdd20180521-4-clm1ue/zerocracy/farm && pdd -v -f /tmp/pdd.xml20180521-1943-184ze1g" took too long and we had to terminate it (execution expired). Most likely your repository is too big for us. Try to ignore unnecessary files by using --exclude option in your .pdd file. More information here: https://github.com/yegor256/pdd#how-to-run.

**Comment by yegor256** [comment]
@g4s8 it can't be too big, since it takes just a few seconds for me to parse it locally on my laptop. There is something else going on, I suspect. Looks like `pdd` just gets stuck and stays and wait forever.

**Comment by yegor256** [comment]
@g4s8 I will try to publish the logs from it to the exception message, somehow. We need to see what it's doing for so long...

**Comment by yegor256** [comment]
@g4s8 I don't know why, but the problem is fixed, thanks to 5af5f085c28fa8f9240021dfb3af77d50b737990

**Comment by yegor256** [comment]
@rultor release, tag is `0.30.2`

**Comment by g4s8** [comment]
@yegor256 yes, it's fixed, thanks

**Comment by yegor256** [comment]
@g4s8 I think the problem was this one. The command line was producing the output and there is a buffer for it. The buffer was full and we were stuck staying and waiting for the buffer to be freed. This never happened. That's why the delay. Now I'm reading from the buffer incrementally. That's why it works. This is my guess.

**Comment by 0crat** [comment]
Order was finished: +30 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#205](https://github.com/yegor256/0pdd/issues/205) is now out of scope
